libraries/ESP8266NetBIOS/ESP8266NetBIOS.cpp

```diff
@@ -212,7 +212,10 @@ void ESP8266NetBIOS::_recv(udp_pcb *upcb, pbuf *pb, const ip_addr_t *addr, uint1
                     nbnsa.NBNSA_TIMETOLIVE = LWIP_PLATFORM_HTONL(300000UL);// Time to live (30000 sekund)
                     nbnsa.NBNSA_LENGTH = LWIP_PLATFORM_HTONS(6);
                     nbnsa.NBNSA_NODEFLAGS = LWIP_PLATFORM_HTONS(0);
-                    nbnsa.NBNSA_NODEADDRESS = WiFi.localIP(); // ulozime nasi IP adresu
+					if (WiFi.getMode()==WIFI_STA)
+						nbnsa.NBNSA_NODEADDRESS = WiFi.localIP(); // ulozime nasi IP adresu
+					else
+						nbnsa.NBNSA_NODEADDRESS = WiFi.softAPIP(); // ulozime nasi IP adresu

```

Comment:
I just realized that it is an old and partial reimplementation of UdpContext, not aware of IPv6.
That is OK because [NetbiosNS/udp137 does not support IPv6](https://en.wikipedia.org/wiki/NetBIOS).

This code is better but still improvable.
Can you please try with
```
nbnsa.NBNSA_NODEADDRESS = ip4_addr_get_u32(ip_2_ip4(ip_current_dest_addr())); // `ip_current_dest_addr()` is usable inside lwIP's UDP receive callback 
```
This address is the IPv4 destination address of the received packet that is replied to, so it is the ESP address on the right interface.
Hence it should work on AP, STA, AP+STA when coming from AP, AP+STA when coming from STA, ethernet, PPP.

Reply:
ok. I will give it a try and let you know...both in esp32 and esp8266 bugs

Reply:
The proposed fix uses lwIP-v2 API that is also used in esp32 Arduino core, and valid whether IPv6 is enabled or not.

Reply:
@pabloandresm Did you have a chance to give it a try ?

Reply:
Hello there.

I've tried the fix with arduino ide esp32 v2.0.2 and v2.0.3 based on ESP-IDF 4.4.1, and it works perfectly.

To sum up, 1st: the fix returns the IP of STA if in STA mode. If not, then it returns the IP of AP mode.

Without the fix, the IP of AP mode is always returned regardless in which mode we are.

Pablo